packages/api/src/graphql/mutation.public.ts

```diff
@@ -9,14 +9,16 @@ import {
   InvalidCredentialsError,
   OAuth2ProviderNotFoundError,
   InvalidOAuth2TokenError,
-  UserNotFoundError
+  UserNotFoundError,
+  NotAuthorisedError as NotAuthorizedError
 } from '../error'
 import {
   GraphQLPublicCommentInput,
   GraphQLPublicCommentUpdateInput,
   GraphQLPublicComment
 } from './comment'
 import {CommentAuthorType, CommentState} from '../db/comment'
+import {UserInputError} from 'apollo-server-express'
```

Comment:
What's the difference between `ApolloError` and `UserInputError`. Also could you move the Error definition to `packages/api/src/error.ts` like the other errors.

Reply:
UserInputError is extended from the base `ApolloError` which can provide more details for the user input data error like the invalid arguments passed. 

I haven't added it to `error.ts` and used it as I found no importance to have an empty new class :S  What do you think?